e2e/cypress/support/api_commands.d.ts

```diff
@@ -188,5 +188,17 @@ declare namespace Cypress {
          *   });
          */
         apiAddUserToChannel(channelId: string, userId: string): Chainable<Response>;
+
+        /**
+         * Save a list of the user's preferences.
+         * See https://api.mattermost.com/#tag/preferences/paths/~1users~1{user_id}~1preferences/put
+         * @param {object[]} preferences - List of preference objects
+         * @param {string} userId - User ID
+         * @returns {Response} response: Cypress-chainable response which should have successful HTTP status of 200 OK to continue or pass.
+         *
+         * @example
+         *   cy.apiSaveUserPreference([{user_id: 'user-id', category: 'display_settings', name: 'channel_display_mode', value: 'full'}], 'user-id');
+         */
+        apiSaveUserPreference(preferences: Array<object>, userId: string): Chainable<Response>;
```

Comment:
You may import [PreferenceType](https://github.com/mattermost/mattermost-redux/blob/master/src/types/preferences.ts) into https://github.com/mattermost/mattermost-webapp/blob/master/e2e/cypress/support/index.d.ts. Then this could be
```suggestion
        apiSaveUserPreference(preferences: PreferenceType[], userId: string): Chainable<Response>;
```

Also, I suggests to move both `apiSaveUserPreference` and `apiSaveClockDisplayMode24HourPreference` commands into its own file, `support/api/preference.js`.

Reply:
Updated